api/src/db/pipelines.rs

```diff
@@ -277,7 +277,7 @@ pub async fn read_all_pipelines(
 /// Helper function to check if an sqlx error is a duplicate pipeline constraint violation
 pub fn is_duplicate_pipeline_error(err: &sqlx::Error) -> bool {
     match err {
-        sqlx::Error::Database(ref db_err) => {
+        sqlx::Error::Database(db_err) => {
```

Comment:
Calling out a few places where upgrading the edition caused compiler errors to make it easy to review. The rest of the noise is mostly due to clippy warning fixes or cargo fmt changes.